pkg/storegateway/bucket_stores.go

```diff
@@ -463,8 +461,8 @@ func (u *BucketStores) getOrCreateStore(userID string) (*BucketStore, error) {
 		userBkt,
 		fetcher,
 		u.syncDirForUser(userID),
-		newChunksLimiterFactory(u.limits, userID),
-		NewSeriesLimiterFactory(0), // No series limiter.
+		NewChunksLimiterFactory(uint64(u.limits.MaxChunksPerQuery(userID))),
+		NewSeriesLimiterFactory(uint64(u.limits.MaxFetchedSeriesPerQuery(userID))),
```

Comment:
Limits can be reloaded at runtime. It's not safe to pass them this way, because they would be read only at startup. You need to pass the function to read limits as we did before with `newChunksLimiterFactory()`.

Reply:
@pracucci: I have reorganized the code and passed the limits via a function. Could you please check again?